intro-dplyr.Rmd

```diff
@@ -0,0 +1,343 @@
+
+
+### Objectives
+
+1. To learn the difference between "messy" and "tidy" data and see the advantages of using data that are consistently structured.
+2. To get to know the tools that help tidying up and reshaping the data.
+3. To perform simple data transformations and create summaries.
+    
+**At the end of this session you will be able:**
+
+1. To perform simple data transformations using dplyr function `mutate()`.
+1. To select variables and observations to work with using dplyr functions `filter()`and `select()`.
+1. To order data by variable using dplyr function `arrange()`.
+1. To provide a simple data summary using dplyr functions `group_by()` and `summarise()`.
+1. To change the structure of the data using tidyr functions `gather()`, `separate()`, and `spread()`.
+
+## Resources for learning and help
+
+**For learning**:
+
+1. "R for Data Science" book: [Chapter on tidy data](http://r4ds.had.co.nz/tidy-data.html)
+2. [tidyr vignette](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html)
+3. [dplyr vignette](https://cran.r-project.org/web/packages/dplyr/vignettes/dplyr.html)
+4. DataCamp course ["Working with Data in the Tidyverse"](https://www.datacamp.com/courses/working-with-data-in-the-tidyverse)
+    - Chapter 1 "Explore your data" is free (the following chapters are available after purchasing the DataCamp subscription)
+
+**For help**:
+    
+1. [Data wrangling cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf)
+2. [StackOverflow for tidyr](https://stackoverflow.com/questions/tagged/tidyr)
+3. [StackOverflow for dplyr](https://stackoverflow.com/questions/tagged/dplyr)
+4. Quick package information using RStudio: `?tidyr`, `?dplyr`
+
+## Quickly look over the data we will use
+
+```{r}
+library(dplyr)
+library(tidyr)
+```
+
+### Exercise: Choose and look at a dataset
+
+Create an `exercise.R` file (we will have done that in class) and put these code into the file.
+
+```{r}
+# Load the packages
+library(dplyr)
+library(tidyr)
+
+# Variables
+names(CO2)
+
+# General contents
+str(CO2)
+
+# Using the dplyr function
+glimpse(CO2)
+
+# Quick summary
+summary(CO2)
+```
+
+
+## "Messy" vs. "tidy" data
+
+**The dataset is tidy when:**
+
+- Each variable has its own column.
+- Each observation has its own row
+- Each value has its own cell
+
+```{r}
+library(tidyr)
+library(dplyr)
+```
+
+Explore the structure of `table1`. Is it tidy?
+
+```{r}
+table1
+```
+
+**Benefits of tidy data and tidy code**:
+
+1. Time spent preparing your data at the beginning can save hours of frustration in the long run.
+2. "Tidy data" provides a concept and the package tidyr provides the functions for this work.
+3. Tibble package makes datasets efficient to work with.
+4. dplyr provides fast and intuitive data processing functions.
+    - For *very very large* datasets, the data.table package is very fast and powerful
+5. The `%>%` pipe operator can help clarify complex data processing workflows.
```

Comment:
You'll probably need a section on just explaining what the pipe is and does. Even for intermediate users, this is a different way of doing R coding.